clusterctl/clusterdeployer/clusterdeployer.go

```diff
@@ -148,39 +151,50 @@ func (d *ClusterDeployer) Create(cluster *clusterv1.Cluster, machines []*cluster
 		return fmt.Errorf("unable to update external cluster endpoint: %v", err)
 	}
 
-	glog.Info("Creating internal cluster")
-	internalClient, err := d.createInternalCluster(externalClient, provider, kubeconfigOutput)
-	if err != nil {
-		return fmt.Errorf("unable to create internal cluster: %v", err)
-	}
-	defer closeClient(internalClient, "internal")
+	targetClient := externalClient
+	targetCluster := "external"
 
-	glog.Info("Applying Cluster API stack to internal cluster")
-	if err := d.applyClusterAPIStackWithPivoting(internalClient, externalClient); err != nil {
-		return fmt.Errorf("unable to apply cluster api stack to internal cluster: %v", err)
-	}
+	if d.pivotCluster {
+		glog.Info("Creating internal cluster")
+		internalClient, err := d.createInternalCluster(externalClient, provider, kubeconfigOutput)
+		if err != nil {
+			return fmt.Errorf("unable to create internal cluster: %v", err)
+		}
+		defer closeClient(internalClient, "internal")
+		targetClient = internalClient
+		targetCluster = "internal"
 
-	glog.Info("Saving provider components to the internal cluster")
-	err = d.saveProviderComponentsToCluster(providerComponentsStoreFactory, kubeconfigOutput)
-	if err != nil {
-		return fmt.Errorf("unable to save provider components to internal cluster: %v", err)
-	}
+		glog.Info("Applying Cluster API stack to internal cluster")
+		if err := d.applyClusterAPIStackWithPivoting(internalClient, externalClient); err != nil {
+			return fmt.Errorf("unable to apply cluster api stack to internal cluster: %v", err)
+		}
+
+		glog.Info("Saving provider components to the internal cluster")
+		err = d.saveProviderComponentsToCluster(providerComponentsStoreFactory, kubeconfigOutput)
+		if err != nil {
+			return fmt.Errorf("unable to save provider components to internal cluster: %v", err)
+		}
 
-	// For some reason, endpoint doesn't get updated in external cluster sometimes. So we
-	// update the internal cluster endpoint as well to be sure.
-	glog.Infof("Updating internal cluster object with master (%s) endpoint", master.Name)
-	if err := d.updateClusterEndpoint(internalClient, provider); err != nil {
-		return fmt.Errorf("unable to update internal cluster endpoint: %v", err)
+		// For some reason, endpoint doesn't get updated in external cluster sometimes. So we
+		// update the internal cluster endpoint as well to be sure.
+		glog.Infof("Updating internal cluster object with master (%s) endpoint", master.Name)
+		if err := d.updateClusterEndpoint(internalClient, provider); err != nil {
+			return fmt.Errorf("unable to update internal cluster endpoint: %v", err)
+		}
+	} else {
```

Comment:
If the `pivotCluster` is false, the `cleanupExternalCluster` needs to be set to `false`. Otherwise, we will provision the cluster API stack and then remove it.

Reply:
This has been addressed here: https://github.com/kubernetes-sigs/cluster-api/blob/b546f32729155b36ae1068b3fef8671d98e209a5/clusterctl/clusterdeployer/clusterdeployer.go#L247